pipelines/main.go

```diff
@@ -113,3 +114,38 @@ func newService(ctx context.Context, basePath string) (*genomics.Service, error)
 func defaultProject() string {
 	return os.Getenv("GOOGLE_CLOUD_PROJECT")
 }
+
+type robustTransport struct {
+	Base http.Transport
+}
+
+func (rt *robustTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	delay := time.Second
+
+	var errors []string
+	for {
+		resp, err := rt.roundTrip(req)
+		if err == nil {
+			return resp, nil
+		}
+		errors = append(errors, fmt.Sprintf("attempt %d: %v", len(errors)+1, err))
+		if len(errors) == 3 {
+			return resp, fmt.Errorf("%d failed requests: %v", len(errors), strings.Join(errors, ", "))
+		}
+
+		delay *= 2
+		time.Sleep(delay)
+	}
+}
+
+func (rt *robustTransport) roundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
@kemp-google Having both `roundTrip` and `RoundTrip` is a bit confusing to read to figure out what's happening just because of the case of a symbol `r`.  It's better if the methods that have better nomenclature that indicate their intended behavior from the point of readability and semantics.